lib/react_on_rails/react_renderer.rb

```diff
@@ -1,20 +1,17 @@
 module ReactOnRails
   class ReactRenderer
 
-    # "this" does not need a closure as it refers to the "this" defined by the
-    # calling the calling context which is the "this" in the execJs environment.
-    def render_js_react_component
-      <<-JS.strip_heredoc
-      function renderReactComponent(componentClass, props) {
-        return this.React.renderToString(
-         componentClass(props)
-        );
-      }
+    # Returns a React element, unwrapping it if the component is a generator function
+    def render_js_react_element(react_component, props)
```

Comment:
Let's change the signature so it's clear that you're not passing the real props.
```
def render_js_react_element(react_component_name, props_name)
```